File: driver-app/lib/data/models/medication.dart
class Medication {
  final int? id;
  final int sosEventId;
  final String name;
  final String? dosage;
  final String? notes;
  final String? createdAt;

  const Medication({
    this.id,
    required this.sosEventId,
    required this.name,
    this.dosage,
    this.notes,
    this.createdAt,
  });

  factory Medication.fromJson(Map<String, dynamic> json) {
    return Medication(
      id: json['id'] as int?,
      sosEventId: json['sosEventId'] as int? ?? 0,
      name: json['name'] as String? ?? '',
      dosage: json['dosage'] as String?,
      notes: json['notes'] as String?,
      createdAt: json['createdAt'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'sosEventId': sosEventId,
      'name': name,
      if (dosage != null) 'dosage': dosage,
      if (notes != null) 'notes': notes,
    };
  }
}

File: backend/src/main/java/com/rakshapoorvak/model/entity/Medication.java
package com.rakshapoorvak.model.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.Instant;

/**
 * Medication administered during an SOS event.
 */
@Entity
@Table(name = "medications")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Medication {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "sos_event_id", nullable = false)
    private SosEvent sosEvent;

    @Column(name = "name", nullable = false, length = 255)
    private String name;

    @Column(name = "dosage", length = 100)
    private String dosage;

    @Column(name = "notes", columnDefinition = "TEXT")
    private String notes;

    @Column(name = "administered_at")
    private Instant administeredAt;

    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @PrePersist
    void onCreate() {
        if (createdAt == null) createdAt = Instant.now();
        if (administeredAt == null) administeredAt = Instant.now();
    }
}
